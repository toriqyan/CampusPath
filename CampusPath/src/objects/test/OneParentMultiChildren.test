# Tests that a graph can be created, four nodes and
# edges from one node to the other three can be added, 
# and four nodes and three edges can be printed out.

# Create a graph
CreateGraph TheLordOfTheRing2

# Add four nodes
AddNode TheLordOfTheRing2 Bilbo
AddNode TheLordOfTheRing2 Gandalf
AddNode TheLordOfTheRing2 Arwin
AddNode TheLordOfTheRing2 Samwise

# Add three edges from Bilbo to different children
AddEdge TheLordOfTheRing2 Bilbo Gandalf TheHobbit
AddEdge TheLordOfTheRing2 Bilbo Arwin TheTwoTowers
AddEdge TheLordOfTheRing2 Bilbo Samwise TheTwoTowers

# Print the nodes in the graph and the outgoing edges from Bilbo
ListNodes TheLordOfTheRing2
ListChildren TheLordOfTheRing2 Bilbo