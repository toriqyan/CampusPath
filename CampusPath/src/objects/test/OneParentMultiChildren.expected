# Tests that a graph can be created, four nodes and
# edges from one node to the other three can be added, 
# and four nodes and three edges can be printed out.

# Create a graph
created graph TheLordOfTheRing2

# Add four nodes
added node Bilbo to TheLordOfTheRing2
added node Gandalf to TheLordOfTheRing2
added node Arwin to TheLordOfTheRing2
added node Samwise to TheLordOfTheRing2

# Add three edges from Bilbo to different children
added edge TheHobbit from Bilbo to Gandalf in TheLordOfTheRing2
added edge TheTwoTowers from Bilbo to Arwin in TheLordOfTheRing2
added edge TheTwoTowers from Bilbo to Samwise in TheLordOfTheRing2

# Print the nodes in the graph and the outgoing edges from Bilbo
TheLordOfTheRing2 contains: Arwin Bilbo Gandalf Samwise
the children of Bilbo in TheLordOfTheRing2 are: Arwin(TheTwoTowers) Gandalf(TheHobbit) Samwise(TheTwoTowers)